core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -116,6 +117,43 @@ public void calculateByRoundRobin(Map<String, String> subnetAzPairs, InstanceGro
         }
     }
 
+    public Map<String, String> filterSubnetByLeastUsedAz(InstanceGroup instanceGroup, Map<String, String> subnetAzPairs) {
+        Map<String, List<String>> azSubnetPairs = createAzSubnetPairsFromSubnetAzPairs(subnetAzPairs);
+        Map<String, Integer> azUsage = new HashMap<>();
+        Set<String> azs = azSubnetPairs.keySet();
+        initializeSubnetUsage(new HashSet<>(subnetAzPairs.values()), azs, azUsage);
```

Comment:
It can be the same, but not always. However, the result will be the same -> every AZ will be set to 0 on the map exactly once

Reply:
I did some refinement here, I don't reuse this block for the AZ init, because is unnecessary. New code: `azs.forEach(az -> azUsage.computeIfAbsent(az, k -> 0));`